src/System.Net.Mail/tests/Unit/MailAddressTests/MailAddressEncodeTest.cs

```diff
@@ -0,0 +1,164 @@
+        {
+            MailAddress testAddress = new MailAddress("test@example.com");
+
+            string result = testAddress.Encode(2, false);
+            Assert.Equal("test@example.com", result);
+
+            result = testAddress.Encode(2, true);
+            Assert.Equal("test@example.com", result);
+        }
+
+        [Fact]
+        public void EncodeSingleMailAddress_WithAddressAndUnicode_AndPaddingValueOfNonZero_ShouldEncodeCorrectly()
+        {
+            MailAddress testAddress = new MailAddress("jeff@nclmailtest.com", "jeff ÃÚêëïï");
+            string result = testAddress.Encode(2, false);
+            Assert.Equal("=?utf-8?Q?jeff_=C3=83=C3=9A=C3=AA=C3=AB=C3=AF=C3=AF?= <jeff@nclmailtest.com>", result);
+
+            result = testAddress.Encode(2, true);
+            Assert.Equal("\"jeff ÃÚêëïï\" <jeff@nclmailtest.com>", result);
+        }
+
+        [Fact]
+        public void EncodeSingleMailAddress_WithAddressOnly_ShouldEncodeAsAddress()
+        {
+            MailAddress testAddress = new MailAddress("test@example.com");
+            string result = testAddress.Encode(0, false);
+            Assert.Equal("test@example.com", result);
+        }
+
+        [Fact]
+        public void EncodeSingleMailAddress_WithAddressAndDisplayName_ShouldEncodeAsDisplayNameAndAddressInBrackets()
+        {
+            MailAddress testAddress = new MailAddress("test@example.com", "test");
+            string result = testAddress.Encode(0, false);
+            Assert.Equal("\"test\" <test@example.com>", result);
+
+            result = testAddress.Encode(0, true);
+            Assert.Equal("\"test\" <test@example.com>", result);
+        }
+
+        [Fact]
+        public void EncodeSingleMailAddress_WithAddressAndDisplayNameUnicode_ShouldQEncode()
+        {
+            MailAddress testAddress = new MailAddress("test@example.com", "testÜ");
+
+            string result = testAddress.Encode(0, false);
+            Assert.Equal("=?utf-8?Q?test=C3=9C?= <test@example.com>", result);
+
+            result = testAddress.Encode(0, true);
+            Assert.Equal("\"testÜ\" <test@example.com>", result);
+        }
+
+        [Fact]
+        public void EncodeSingleMailAddress_WithAddressAndLongDisplayNameUnicode_ShouldQEncode()
+        {
+            MailAddress testAddress = new MailAddress("test@example.com", 
+                "testÜtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest");
+            string result = testAddress.Encode(0, false);
+            Assert.Equal("=?utf-8?Q?test=C3=9Ctesttesttesttesttesttesttesttesttesttesttesttest?=\r\n "
+                + "=?utf-8?Q?testtesttesttesttesttest?= <test@example.com>", result);
+            
+            result = testAddress.Encode(0, true);
+            Assert.Equal("\"testÜtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest\" "
```

Comment:
@Priya91 Same here: please escape Unicode. (Also other places - will stop adding this comment.)